skimage/restoration/inpaint.py

```diff
@@ -121,16 +123,27 @@ def inpaint_biharmonic(img, mask, multichannel=False):
     img = skimage.img_as_float(img)
     mask = mask.astype(np.bool)
 
+    # Split inpainting mask into independent regions
+    kernel = ndi.morphology.generate_binary_structure(mask.ndim, 1)
+    kernel = ndi.iterate_structure(kernel, 2)
+    mask_dilated = ndi.morphology.binary_dilation(mask, structure=kernel)
```

Comment:
>Won't this combine independent, adjacent sub-regions?

That's exactly the reason behind. See the details under @jni's comment.

As for big images, my feeling is that a single dilation run is nearly nothing compared to benefit of running `spsolve` for each region separately. Of course, for some special cases, this could not be the case (batch processing of 10000x10000 images with a single inpainting region per image), but, in general case, I expect this to be a good change.
If you insist, we could make this optional via addition argument.